### PR TITLE
Fix UndoRedo when creating new AnimationStateMachine transition and node

### PR DESCRIPTION
**Pull request by SaracenOne**
Fixes INVALID_HISTORY UndoRedo error when creating a new transition which in turn creates a new node. Was caused by two seperate actions being committed and fixed by properly handling the p_nested_action argument.

### CODE REVIEW

File: editor/plugins/animation_state_machine_editor.cpp [hunk 1]
```diff
@@ -1565,6 +1561,7 @@ void AnimationNodeStateMachineEditor::_notification(int p_what) {
 			transition_mode->add_icon_item(get_theme_icon(SNAME("TransitionImmediate"), SNAME("EditorIcons")), TTR("Immediate"));
 			transition_mode->add_icon_item(get_theme_icon(SNAME("TransitionSync"), SNAME("EditorIcons")), TTR("Sync"));
 			transition_mode->add_icon_item(get_theme_icon(SNAME("TransitionEnd"), SNAME("EditorIcons")), TTR("At End"));
+			transition_mode->add_icon_item(get_theme_icon(SNAME("TransitionTime"), SNAME("EditorIcons")), TTR("At Time"));
```
Comment (akien-mga) [inline editor/plugins/animation_state_machine_editor.cpp:1564]: Seems like an unrelated change that crept up from other work? I don't see `SWITCH_MODE_AT_TIME` in `AnimationNodeStateMachineTransition::SwitchMode`.

Comment (SaracenOne) [reply editor/plugins/animation_state_machine_editor.cpp:1564]: Yes, that got accidentally committed from some unrelated experimental work I was doing. I shall remove it now 👀

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
